contracts/test-contracts/MockContract.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+contract MockContract {
+    event Called(uint256 value, bytes data);
+
+    struct Result {
+        bool failure;
+        bytes returnData;
+    }
+
+    // The mapping from calldata to call result, will return empty return data with successful result by default.
+    mapping(bytes => Result) results;
+
+    // This function call will not pass to fallback, but this is fine for the tests.
+    function setResult(bytes calldata _calldata, Result calldata result) external {
+        results[_calldata] = result;
+    }
+
+    fallback() external payable {
+        uint256 len;
+        assembly {
+            len := calldatasize()
+        }
+        bytes memory data = new bytes(len);
+        assembly {
+            calldatacopy(add(data, 0x20), 0, len)
+        }
+
+        bool failure = results[data].failure;
+        bytes memory returnData = results[data].returnData;
+
+        // Most likely that's some call to perform some action, so context is not static.
+        if (!failure && returnData.length == 0) {
+            emit Called(msg.value, data);
```

Comment:
this is just for debugging, right?

Reply:
We are checking that this event was emitted sometimes in tests. For example, to check that known codes storage calls l1 messenger during the `markBytecodeAsPublished` function execution

Reply:
I mean, we are emitting only if an empty result is expected, and it's convenient to check it. If we don't need it - we can ignore this event. Do you think it will affect performance?

Reply:
I dont think it affects performance, just thought that for consistency it might be better to always emit it. In any case, not a big deal so up to you. In case anyone will ever always need this event, they can add it

Reply:
Ok, I understood you know. So we can not emit it always, because this mock is used for view methods as well. So in the static context, it will just fail. At least I should add better comments.